plugins/provisioners/puppet/config/puppet.rb

```diff
@@ -35,22 +34,30 @@ def initialize
           @module_path           = UNSET_VALUE
           @options               = []
           @facter                = {}
-          @synced_folder_type    = UNSET_VALUE
+          @synced_folder_opts    = {}
           @temp_dir              = UNSET_VALUE
           @working_directory     = UNSET_VALUE
           @structured_facts   = UNSET_VALUE
         end
 
-        def nfs=(value)
-          puts "DEPRECATION: The 'nfs' setting for the Puppet provisioner is"
-          puts "deprecated. Please use the 'synced_folder_type' setting instead."
-          puts "The 'nfs' setting will be removed in the next version of Vagrant."
+        def synced_folder_type=(value)
+          puts "DEPRECATION: The 'synced_folder_type' setting for the Puppet provisioner is"
+          puts "deprecated. Please use the 'synced_folder_opts' setting instead."
+          puts "The 'synced_folder_type' setting will be removed in the next version of Vagrant."
```

Comment:
This text should be abstracted out into our string translations file with `I18n`. If you're not familiar with that, essentially you copy this and add a I18n "key" to be referenced later in the `en.yml` file (I have an example below). Also the warning should be printed in the `validation` method so that we can used the passed in `machine` object to properly print the warning like so:

```ruby
def validate(machine)
...
  if @synced_folder_type
    machine.ui.warn(I18n.t("vagrant.provisioners.puppet.synced_folder_type_deprecation"))
  end
...
end
```

And same goes for the `synced_folder_args` method below!

Also then because it's just setting a value to this option, we don't need a special method for it so they can go back to how they were prior to this PR. And you'll need to add @synced_folder_args as a value being initialized with `UNSET_VALUE` too (It looks like it didn't exist before...).

Reply:
Sure, this looks very reasonable to me and i will follow your advice.